sound/soc/intel/boards/sof_sdw_cs_amp.c

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Copyright (c) 2023 Intel Corporation
+
+/*
+ *  sof_sdw_cs_amp - Helpers to handle CS35L56 from generic machine driver
+ */
+
+#include <linux/device.h>
+#include <linux/errno.h>
+#include <sound/soc.h>
+#include <sound/soc-acpi.h>
+#include "sof_sdw_common.h"
+
+#define CODEC_NAME_SIZE	8
+
+static int cs_spk_init(struct snd_soc_pcm_runtime *rtd)
+{
+	const char *dai_name = rtd->dai_link->codecs->dai_name;
+	struct snd_soc_card *card = rtd->card;
+	char codec_name[CODEC_NAME_SIZE];
+
+
+	snprintf(codec_name, CODEC_NAME_SIZE, "%s", dai_name);
+	card->components = devm_kasprintf(card->dev, GFP_KERNEL,
+					  "%s spk:%s",
+					  card->components, codec_name);
```

Comment:
why we need to first copy dai_name to codec_name, then copy codec_name to component name ? can we do it as
````c
card->components = devm_kasprintf(card->dev, GFP_KERNEL,
					  "%s spk:%s",
					  card->components, dai_name);
````

Reply:
> why we need to first copy dai_name to codec_name, then copy codec_name to component name ? can we do it as
> 
> ```c
> card->components = devm_kasprintf(card->dev, GFP_KERNEL,
> 					  "%s spk:%s",
> 					  card->components, dai_name);
> ```

codec_name is a subset of dai_name. We only want to take the first CODEC_NAME_SIZE chars.